src/m365/planner/commands/task/task-reference-remove.ts

```diff
@@ -0,0 +1,141 @@
+import * as _ from 'lodash';
+import { Cli, Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { formatting } from '../../../../utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url?: string;
+  alias?: string;
+  taskId: string;
+  confirm?: boolean;
+}
+
+class PlannerTaskReferenceRemoveCommand extends GraphCommand {  
+  public get name(): string {
+    return commands.TASK_REFERENCE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the reference from the Planner task';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.confirm) {
+      this.removeReference(logger, args, cb);
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the reference from the Planner task?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.removeReference(logger, args, cb);
+        }
+      });
+    }
+  }
+
+  private removeReference(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getTaskDetailsEtagAndUrl(args.options)
+      .then(({etag, url}) => {
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            references: {
+              [formatting.openTypesEncoder(url)]: null
+            }
+          }
+        };
+
+        return request.patch(requestOptionsTaskDetails);
+      })
+      .then((res: any): void => {
+        logger.log(res.references);
```

Comment:
Since we're returning the API's response, we should define defaultProperties. Otherwise, in the text output, the response is unreadable:

<img width="1125" alt="image" src="https://user-images.githubusercontent.com/11164679/171572244-59937b86-9d70-41a5-ad77-ec0e6ba7ece4.png">

Reply:
We can do that, indeed. If someone needs the output, they could use the list command. If need be, we can always consider adding it back in the future.